src/index.ts

```diff
@@ -110,37 +140,60 @@ function defineAuthorFactoryInternal<TOptions extends AuthorFactoryDefineOptions
  * @param options
  * @returns factory {@link AuthorFactoryInterface}
  */
-export function defineAuthorFactory<TOptions extends AuthorFactoryDefineOptions>(
+export function defineAuthorFactory<TOptions extends AuthorFactoryDefineOptions<{}>>(
   options: TOptions,
-): AuthorFactoryInterface<TOptions> {
-  return defineAuthorFactoryInternal(options);
+): AuthorFactoryInterface<{}, TOptions> {
+  return defineAuthorFactoryInternal({}, options);
+}
+
+/**
+ * Define factory for {@link Author} model with transient fields.
+ *
+ * @param options
+ * @returns factory {@link AuthorFactoryInterface}
+ */
+export function defineAuthorFactoryWithTransientFields<
```

Comment:
I really wanted to allow `transientFields` in the `defineAuthorFactory`, but I couldn't. For more information: #17